extension/scripts/features/enemy-filter/ttEnemyFilter.js

```diff
@@ -28,32 +29,38 @@
 
 			filterRow(row, { statsEstimates }, true);
 		});
-		addXHRListener(async ({ detail: { page, xhr } }) => {
-			if (!feature.enabled()) return;
-			if (page !== "userlist") return;
 
-			const step = new URLSearchParams(xhr.requestBody).get("step");
-			if (step !== "blackList") return;
+		const listObserver = new MutationObserver((mutations) => {
+			if (mutations.some((mutation) => [...mutation.addedNodes].some((node) => node.matches("li[class*='tableRow__']")))) {
+				if (filterSetupComplete && feature.enabled())
+					applyFilters(true);
+			}
+		});
 
-			new MutationObserver((mutations, observer) => {
-				addFilters();
-				observer.disconnect();
-			}).observe(document.find(".blacklist"), { childList: true });
+		const tableObserver = new MutationObserver((mutations) => {
+			if (mutations.some((mutation) => [...mutation.addedNodes].some((node) => node.tagName === "UL"))) {
+				if (filterSetupComplete && feature.enabled()) {
+					applyFilters(true);
+					listObserver.observe(document.find(".tableWrapper > ul"));

```

Comment:
Uncaught TypeError: Failed to execute 'observe' on 'MutationObserver': The options object must set at least one of 'attributes', 'characterData', or 'childList' to true.